File: app/src/main/java/com/pixplicity/cryptogram/views/CryptogramLayout.java
package com.pixplicity.cryptogram.views;

import android.annotation.SuppressLint;
import android.content.Context;
import android.support.annotation.Nullable;
import android.util.AttributeSet;
import android.view.MotionEvent;
import android.widget.LinearLayout;


public class CryptogramLayout extends LinearLayout {

    private CryptogramView mCryptogramView;

    public CryptogramLayout(Context context) {
        super(context);
        init(context);
    }

    public CryptogramLayout(Context context, @Nullable AttributeSet attrs) {
        super(context, attrs);
        init(context);
    }

    public CryptogramLayout(Context context, @Nullable AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
        init(context);
    }

    private void init(Context context) {
        setClickable(true);
    }

    @SuppressLint("ClickableViewAccessibility")
    @Override
    public boolean onTouchEvent(MotionEvent event) {
        switch (event.getAction()) {
            case MotionEvent.ACTION_UP:
                if (mCryptogramView != null) {
                    mCryptogramView.hideSoftInput();
                    // Clear selected character
                    mCryptogramView.setSelectedCharacter((char) 0);
                }
                break;
        }
        return super.onTouchEvent(event);
    }

    public void setCryptogramView(CryptogramView view) {
        mCryptogramView = view;
    }

}

File: app/src/main/java/com/pixplicity/cryptogram/views/CryptogramLayout.kt
package com.pixplicity.cryptogram.views

import android.annotation.SuppressLint
import android.content.Context
import android.util.AttributeSet
import android.view.MotionEvent
import android.widget.LinearLayout


class CryptogramLayout : LinearLayout {

    private var mCryptogramView: CryptogramView? = null

    constructor(context: Context) : super(context) {
        init(context)
    }

    constructor(context: Context, attrs: AttributeSet?) : super(context, attrs) {
        init(context)
    }

    constructor(context: Context, attrs: AttributeSet?, defStyleAttr: Int) : super(context, attrs, defStyleAttr) {
        init(context)
    }

    private fun init(context: Context) {
        isClickable = true
    }

    @SuppressLint("ClickableViewAccessibility")
    override fun onTouchEvent(event: MotionEvent): Boolean {
        when (event.action) {
            MotionEvent.ACTION_UP -> if (mCryptogramView != null) {
                mCryptogramView!!.hideSoftInput()
                // Clear selected character
                mCryptogramView!!.selectedCharacter = 0.toChar()
            }
        }
        return super.onTouchEvent(event)
    }

    fun setCryptogramView(view: CryptogramView) {
        mCryptogramView = view
    }

}
